akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ClusterSharding.scala

```diff
@@ -761,7 +761,7 @@ private[akka] class ClusterShardingGuardian extends Actor {
             val singletonSettings = settings.coordinatorSingletonSettings.withSingletonName("singleton").withRole(role)
```

Comment:
In above `BackoffOpts` we must also add
```
.withFinalStopMessage(msg => msg ==ShardCoordinator.Internal.Terminate)
```

otherwise it's possible that the Terminate message is not handled if the coordinator has already been stopped (e.g. due to persist failure)

Reply:
You are right! I added withFinalStopMessage